File: baekjoon/String/1769.cpp
// 백준 1769번 : 3의 배수
#include <iostream>

using namespace std;

int main()
{
    int ans = 0;
    string num;
    cin >> num;
    
    while (num.length() > 1) {
        int sum = 0;
        for (int i = 0; i < num.length(); i++) {
            sum += num[i] - '0';
        }
        num = to_string(sum);
        ans++;
    }
    
    cout << ans << '\n';
    if (stoi(num) % 3 == 0) {
        cout << "YES";
    } else {
        cout << "NO";
    }
    return 0;
}

File: baekjoon/String/1769.py
# 백준 1769번 : 3의 배수

def func(num, cnt):
    if len(num) > 1:
        sum = 0
        for ch in num:
            sum += int(ch)
        func(str(sum), cnt + 1)

    elif len(num) == 1:
        if int(num) % 3 == 0:
            print(cnt)
            print("YES")
        else:
            print(cnt)
            print("NO")

if __name__ == "__main__":
    string = str(input())
    func(string, 0)